R/bundle.R

```diff
@@ -60,71 +60,88 @@ isKnitrCacheDir <- function(subdir, contents) {
   }
 }
 
-maxDirectoryList <- function(dir, parent, totalSize) {
+# dir is the path for this step on our recursive walk.
+# topLevel is TRUE only for the outermost directory of our walk.
+# totalSize is a running total of our encountered file sizes.
+# totalFiles is a running count of our encountered files.
+maxDirectoryList <- function(dir, topLevel, totalFiles, totalSize) {
```

Comment:
One potential improvement: what about using `depth` rather than `topLevel` here, just to track how far we're recursing? We could also take this as an opportunity to bail if we saw we were recursing excessively.